var/spack/repos/builtin/packages/py-charm4py/package.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+import os.path
+
+
+class PyCharm4py(PythonPackage):
+    """Charm4py (Charm++ for Python) is a distributed computing and parallel
+    programming framework for Python, for the productive development of fast,
+    parallel and scalable applications. It is built on top of Charm++, a C++
+    adaptive runtime system that has seen extensive use in the scientific and
+    high-performance computing (HPC) communities across many disciplines, and
+    has been used to develop applications that run on a wide range of devices:
+    from small multi-core devices up to the largest supercomputers."""
+
+    homepage = "https://charmpy.readthedocs.io"
+    url = "https://pypi.io/packages/source/c/charm4py/charm4py-1.0.tar.gz"
+
+    # Add a list of GitHub accounts to
+    # notify when the package is updated.
+    maintainers = ['payerle']
+
+    # Get errors passing --mpi to build* phases of setup.py
+    phases = ['install']
+
+    version('1.0', sha256='8ddb9f021b7379fde94b28c31f4ab6a60ced2c2a207a2d75ce57cb91b6be92bc')
+
+    variant('mpi', default=True,
+            description='build Charm++ library with the MPI instead of TCP'
+            ' communication layer')
+
+    # Builds its own charm++, so no charmpp dependency
+    depends_on('py-greenlet')
+    depends_on('py-cython')
+    depends_on('cuda')
+    depends_on('mpi', when='+mpi')
+
+    # setup.py builds its own charm++, but libcharm.so
+    # ends up with a cuda dependency causing unresolved symbol errors
+    # when setup.py tries to load it to get version.  We need to explicitly
+    # link libcudart when building the charm++ library.
+    # To do this, the following patch:
+    # 1) hacks setup.py to apply a patch to the charm++ Makefile
+    # causing the Makefile to include libcudart when building libcharm.so
+    # 2) inserts the patchfile needed to do so.
+    # This is convoluted, but best way I see since setup.py untars the
+    # charm++ sources and we need to patch a file that is in the tarball.
```

Comment:
Hmm, this is rather convoluted. How hard would it be to build against an external `charm++` installation? That would cut down on build times for users who need both.